api/prometheus/v1/api_test.go

```diff
@@ -705,7 +714,11 @@ func TestAPIs(t *testing.T) {
 		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
 			client.curTest = test
 
-			res, err := test.do()
+			res, warnings, err := test.do()
+
+			if (test.inWarnings == nil) != (warnings == nil) {
```

Comment:
I am a bit confused by this. The `queryTests` don't have the `inWarnings` set anywhere so this will always be true `test.inWarnings == nil`

Reply:
@krasi-georgiev true, I've added a couple cases that do so.

Reply:
but than you should also check the content right? `testutil.Equals(t,test.inWarnings,warnings)`

Reply:
and don't forget the full stop for the comments to follow the golang standards. I know it is annoying until you get used to it :smile_cat:

Reply:
Updated again :)